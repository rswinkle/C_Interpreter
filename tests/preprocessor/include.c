#include "../tests/inc_test1.c"
a
/*
#ifndef FIB_NUM
#define FIB_NUM 5
#endif
*/


int main()
{
	int local = 54
	print local;
	func1();
	print a;
	func2(10, local /= 3, local - a*20);
	print local;

	local = fibbonaci(FIB_NUM);
	print local;

	return 0;
}
