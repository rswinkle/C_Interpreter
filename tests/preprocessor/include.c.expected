# 0 "../tests/preprocessor/include.c"
# 0 "../tests/inc_test1.c"


int a = 1;
# 0 "../tests/inc_test2.c"

void func1(void)
{
	print a;
	int a = 2;
	print a;
	int local = 10 * a;
	print local;
}
# 4 "../tests/inc_test1.c"

void func2(int a, int b, int c)
{
	print a;
	print b;
	print c;

	int d = 12;
	{
		print d;
		print a;
		int a = 15;
		print a;
		{
			print d;
			print a;
			print b;

			int b = 100;
			print b;
			print c;
			{
				print c;
			}

		}
		int c = 5;
		print c;

	}
}



int fibbonaci(int n)
{
	print n;
	if (n < 3) {
		return 1;
	}
	int a = n + 2;
	print a;
	return fibbonaci(n-2) + fibbonaci(n-1);
}



# 1 "../tests/preprocessor/include.c"
a



int main()
{
	int local = 54
	print local;
	func1();
	print a;
	func2(10, local /= 3, local - a*20);
	print local;

	local = fibbonaci(4);
	print local;

	return 0;
}

